src/components/panels/UnitsPanel/UnitsPanel.js

```diff
@@ -1,24 +1,63 @@
 import React, { useMemo, useCallback } from "react";
 import SlidePanel from "components/SlidePanel/SlidePanel";
 import MainTable from "@canonical/react-components/dist/components/MainTable";
+import cloneDeep from "clone-deep";
 
 import useModelStatus from "hooks/useModelStatus";
 
 import {
   machineTableHeaders,
   applicationTableHeaders,
+  generateMachineRows,
+  generateApplicationRows,
 } from "pages/Models/Details/generators";
 
 import { generateStatusElement, extractRevisionNumber } from "app/utils";
 
 import "./_units-panel.scss";
 
-export default function UnitsPanel({ isActive, onClose, entity: unitId }) {
+export default function UnitsPanel({
+  isActive,
+  onClose,
+  entity: unitId,
+  panelRowClick,
+}) {
   const modelStatusData = useModelStatus();
   const appName = unitId?.split("/")[0];
   const unit = modelStatusData?.applications[appName]?.units[unitId];
   const app = modelStatusData?.applications[appName];
 
+  const filteredModelStatusDataByMachine = useCallback(
```

Comment:
Doing this all the time is very inefficient, we should move this into an optimized selector.

Reply:
This is for a follow-up sometime, not for now :)